src/components/Modal/Modal.js

```diff
@@ -75,79 +84,83 @@ class Modal extends Component {
     if (prevProps.show !== show) {
       this.toggle()
     }
   }
 
   render = () => {
     const {
-      ariaDescribedby,
-      ariaLabelledby,
-      cancelClick,
+      id,
+      title,
+      type,
+      show,
+      onClose,
+      onConfirm,
+      onCancel,
+      confirmText,
       cancelText,
-      children,
-      className,
       confirmAnalyticsString,
-      confirmClick,
-      confirmText,
-      dataId,
-      footer,
-      id,
       idString,
-      onHide,
-      type,
-      variant,
+      ariaLabelledby,
+      ariaDescribedby,
+      dataId,
+      children,
+      hasCloseButton,
       ...props
     } = this.props
 
     return (
-      <div ref={this.mainRef}>
-        {/* needed for close event from crashing. may need a better solution */}
-        <div data-sprk-modal-trigger={id} />
-        <div
-          className={this.className}
-          ref={this.modalRef}
-          role='dialog'
-          tabIndex='-1'
-          aria-labelledby={ariaLabelledby}
-          aria-modal='true'
-          aria-describedby={ariaDescribedby}
-          data-sprk-modal={id}
-          data-id={dataId}
-          {...props}
-        >
-          <div className='sprk-o-Stack sprk-o-Stack--large'>
-            {type === MODAL_VARIANTS.INFO && (
-              <InfoContent
-                modalName={id}
-                body={children}
-                close={() => this.hide()}
-              />
-            )}
-            {type === MODAL_VARIANTS.CHOICE && (
-              <ChoiceContent
-                modalName={id}
-                body={children}
-                close={() => this.hide()}
-              />
-            )}
-            {type === MODAL_VARIANTS.WAIT && (
-              <WaitContent
+      <div className='sprk-u-JavaScript'>
+        <div data-sprk-modal-trigger={id} />  {/* This needs to be fixed, their method looks for this attribute and sets focus to it */}
+        <div ref={this.mainRef}>
+          <div
+            className={this.className}
+            ref={this.modalRef}
+            role='dialog'
+            tabIndex='-1'
+            aria-labelledby={ariaLabelledby}
+            aria-modal='true'
+            aria-describedby={ariaDescribedby}
+            data-sprk-modal={id}
+            data-sprk-modal-type={type}
+            data-id={dataId}
+            {...props}
+          >
+            <Stack itemSpacing={'large'}>
+              <ModalHeader
+                id={ariaLabelledby}
+                onClose={onClose}
+                title={title}
                 modalName={id}
-                body={children}
-                close={() => this.hide()}
+                hasCloseButton={type !== MODAL_VARIANTS.WAIT ? hasCloseButton : false}
               />
-            )}
+              <ModalBody>
+                {type === MODAL_VARIANTS.WAIT && <Spinner />}
+                <p className='sprk-o-Stack__item sprk-b-TypeBodyTwo' id={ariaDescribedby}>
```

Comment:
part of the reason the contents are split out into a diff component is so the class name can be generated there and not hard coded. also with time this may change and we may add more functionality and it will leave this component easier to follow.
Another option is to create a modal wrapper and create 3 different modals, but that would require a larger refactor, but that would be even cleaner

Reply:
should we just remove the <p> attribute and leave it as children? This would allow us to essentially pass anything to the modal body.

Reply:
yep, that works and it's more flexible